File: src/values/u64.rs
pub struct ValueU64 {
    value: Option<u64>,
}

impl ValueU64 {
    pub fn init_empty() -> ValueU64 {
        ValueU64 { value: None }
    }
    
    pub fn init_with(value: u64) -> ValueU64 {
        ValueU64 { value: Some(value) }
    }
    
    pub fn show(&self, end: char) {
        if let Some(val) = self.value {
            print!("{val}{end}");
        } else {
            print!("null{end}");
        }
    }
    
    pub fn is_null(&self) -> bool {
        return self.value.is_none();
    }
    
    pub fn get_type_representation(&self) -> &str {
        return "u64";
    }
    
    pub fn get_value(&self) -> Option<u64> {
        return self.value
    }
}

File: src/values/u64.cpp
#include "values/u64.hpp"
using namespace common::values;

ValueU64::ValueU64() {
    this->value = std::nullopt;
}

ValueU64::ValueU64(uint64_t value) {
    this->value = value;
}

void ValueU64::show(char end) const {
    if (this->value)
        std::cout << "u64<" << this->value.value() << '>' << end;
    else
        std::cout << "u64<null>" << end;
}

bool ValueU64::isNull() const {
    return !this->value.has_value();
}

std::string ValueU64::getTypeRepr() const {
    return "u64";
}

ValueKind ValueU64::getKind() const {
    return ValueKind::U64;
}

std::optional<uint64_t> ValueU64::getValue() const {
    return this->value;
}